charts/kube-state-metrics/templates/deployment.yaml

```diff
@@ -119,23 +124,25 @@ spec:
         {{- else }}
         image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
         {{- end }}
+        {{- if eq .Values.kubeRBACProxy.enabled false }}
```

Comment:
Don't create the port mapping from the state-metrics container if rbac-proxy is enabled